src/detection/HeavyFootprint.cc

```diff
@@ -333,6 +334,14 @@ typename HeavyFootprint<ImagePixelT, MaskPixelT, VariancePixelT>::Factory
 //
 //
 #define INSTANTIATE(TYPE)                                                                                  \
+    } /* namespace detection */                                                                            \
+    namespace table {                                                                                      \
+    namespace io {                                                                                         \
+    template std::shared_ptr<detection::HeavyFootprint<TYPE>>                                              \
+    PersistableFacade<detection::HeavyFootprint<TYPE>>::dynamicCast(std::shared_ptr<Persistable> const&);  \
+    } /* namespace io */                                                                                   \
+    } /* namespace table */                                                                                \
+    namespace detection {                                                                                  \
```

Comment:
This seems more than a little bug prone, especially the initial closing brace. Can you make `INSTANTIATE` context-independent (e.g., declare `::lsst::table::io::PersistableFacade`)? Or allow argument-dependent lookup?

Reply:
Based on further comments below you are upset about having namespace changes inside of macro boundaries. I am also uncomfortable with this. Unfortunately I could not figure out how to avoid it -- the naive things I tried (e.g. providing full namespaces) didn't work. It appears that the symbols have to be defined in the namespace they are to inhabit.

However, it might make things a bit less ugly if I managed the namespaces fully within the macro -- start from nothing and end with nothing.